ui/components/src/TimeSeriesTooltip/TimeChartTooltip.tsx

```diff
@@ -63,7 +65,12 @@ export const TimeChartTooltip = memo(function TimeChartTooltip({
   const chartWidth = chart?.getWidth() ?? FALLBACK_CHART_WIDTH; // Fallback width not likely to ever be needed.
 
   const containerElement = containerId ? document.querySelector(containerId) : undefined;
-  const cursorTransform = assembleTransform(mousePos, chartWidth, pinnedPos, height ?? 0, width ?? 0, containerElement);
+  // if tooltip is attached to a container, set max height to the height of the container so tooltip does not get cut off
+  const maxHeight = containerElement ? containerElement.getBoundingClientRect().height : undefined;
```

Comment:
Nice! This is a really cool solution and I like that it's opt-in so if containerId isn't set